index.js

```diff
@@ -436,13 +436,11 @@ function processStyle(glStyle, map, baseUrl, host, path, accessToken) {
           layer = setupGeoJSONLayer(glSource, path);
         }
         glSourceId = id;
-      }
-      layerIds.push(glLayer.id);
-      if (layer) {
         layer.setZIndex(i);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') TypeError can be thrown as "layer" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AS2259)